app/assets/javascripts/legislation_annotatable.js

```diff
@@ -226,6 +226,21 @@
           });
         });
       });
+
+      $("body").on("click", ".comment-box a.publish-comment", function(e) {
+        e.preventDefault();
+        var annotation_id = $(this).closest(".comment-box").data("id");
+        $("a.publish-comment").hide();
+        $("#js-comment-form-annotation-" + annotation_id).toggle();
+        $("#js-comment-form-annotation-" + annotation_id + " textarea").trigger("focus");
+        return;
+      });
+    },
+    destroy: function() {
+      if ($(".legislation-annotatable").length === 0) {
```

Comment:
Yes, its needed as this method is executed on `turbolinks:before-cache` event so each time user leaves a page without Annotator we willl have a  JS error because `app` is not defined. I will update to your simplified version. 👍

Reply:
What do you think about next variant @javierm ? It's even shorter

```
if ($(".legislation-annotatable").length > 0) {
  App.LegislationAnnotatable.app.destroy();
}
```

Reply:
> Yes, its needed as this method is executed on turbolinks:before-cache event so each time user leaves a page without Annotator we willl have a JS error because app is not defined.

Thanks for the explanation!

> `if ($(".legislation-annotatable").length > 0) {`

:+1: